### ISSUE TITLE
Table conversion fail when containing <h> tags

### ISSUE THREAD

**Issue by tinylittlehoneybunnytuttifuttipumpkinpi** [breakdance/breakdance#7]
Breakdance is unable to convert tables that contain text with <h> heading tags.
Example:
======================
<table style="width:100%">
  <tr>
    <th><h3>Firstname</h3></th>
    <th>Lastname</th> 
    <th>Age</th>
  </tr>
</table>
======================
(this converts to markdown that looks identical to the above input)

This behavior seems partially correct since <h> tags are not supported by tables.
However, total failure of conversion to markdown cannot be justified simply because tables can contain any text like '#%&@#&!", so any unsupported or unrecognizable syntax should also be treated like text.
Therefore the expected output should be:
======================
| <h3>Firstname</h3> | Lastname | Age |
======================

The problem is further amplified by the fact that currently Breakdance leaves the whole <table> HTML syntax in the markdown output, thus making it no longer usable as markdown. At minimum any bad syntax could be removed or replaced with a warning like:
[Unsupported HTML syntax - cannot be converted to markdown]

**Comment by jonschlinkert** [comment]
> The problem is further amplified by the fact that currently Breakdance leaves the whole <table>HTML syntax in the markdown output, thus making it no longer usable as markdown.

Are there nested tables?

**Comment by jonschlinkert** [comment]
> thanks Breakdance developers for your great contributions!

no prob, stars are always appreciated!

**Comment by littlehoneybunnytuttifruttipumpkinpie** [comment]
> Are there nested tables?

Nope. (the above table example is the simplest that doesn't work)

**Comment by jonschlinkert** [comment]
> At minimum any bad syntax could be removed or replaced with a warning like

Yeah, that would make sense if it was intentional lol. Looks like a bug. I'll look into it, or a pr would be great. here is the [table logic](https://github.com/breakdance/breakdance/blob/master/lib/tables.js)

**Comment by littlehoneybunnytuttifruttipumpkinpie** [comment]
Oh, my JS knowledge is so basic, I have trouble looking through that :-)
But, I see some code that specifically checks for `<h>` tags inside the table?
```
if (/<h[1-6][^>]*>/.test(node.html)) {
        this.emit('\n', node);
        this.emit(node.html);
        return;
      }
```

**Comment by littlehoneybunnytuttifruttipumpkinpie** [comment]
A possible solution could be to actually convert `<h>` to markup within the table before processing the table syntax. Or maybe convert `<h[1-6]>` to `<b>` as a workaround, before parsing the table. Developers could still replace that code with something else if they don't like the `<b>`.

**Comment by littlehoneybunnytuttifruttipumpkinpie** [comment]
(I can probably do this via the "preprocess" event though)
